blockdag/reachability.go

```diff
@@ -155,7 +155,7 @@ func exponentialFractions(sizes []uint64) []float64 {
 // property of reachability intervals that intervals are either completely disjoint,
 // or one strictly contains the other.
 func (ri *reachabilityInterval) isAncestorOf(other *reachabilityInterval) bool {
-	return ri.start <= other.end && other.end <= ri.end
+	return ri.start <= other.end && other.end < ri.end
```

Comment:
Pretty sure that if `ri.start < other.end && other.end <= ri.end` we should also return true.

In addition to fix, please make sure we have test cases for all:
1. ri == other
2. ri.start == other.start && ri.end != other.end
3. ri.end != other.end && ri.end == other.end
4. ri.start > other.start && ri.end < other.end
5. ri.start < other.start && ri.end > other.end